CHANGELOG.md

```diff
@@ -57,6 +57,7 @@
 - Clean up timing of layout tree flattening/ copying of unflattened tree for Weaver. [Michael Zuccarino](https://github.com/mikezucc) [#1157](https://github.com/TextureGroup/Texture/pull/1157)
 - Fix crash setting attributed text on multiple threads [Michael Schneider](https://github.com/maicki)
 - [ASTextNode2] Ignore certain text alignments while calculating intrinsic size [Huy Nguyen](https://github.com/nguyenhuy) [#1166](https://github.com/TextureGroup/Texture/pull/1166)
+- [ASImageNode] Fix a threading issue which can cause a display completion block to never be executed. [Huy Nguyen](https://github.com/nguyenhuy) [#1148](https://github.com/TextureGroup/Texture/pull/1148)
```

Comment:
From the header file, it isn't too clear that calling the method multiple times will overwrite the previous block without calling it. In order to guarantee balanced calls, we'd probably need to track them with an array.

This seems like an increasingly common case in the framework (even interfaceStateDelegates now), so it might be worth creating a helper class like ASBlockList / ASBlockQueue to manage efficiently. Not important right now, just an idea.

Reply:
Agree that it'd be nice to have track all blocks in an array.